internal/cluster/distribution/eks/eksworkflow/activity_update_node_group.go

```diff
@@ -168,6 +169,10 @@ func (a UpdateNodeGroupActivity) Execute(ctx context.Context, input UpdateNodeGr
 			ParameterKey:     aws.String("NodeSecurityGroup"),
 			UsePreviousValue: aws.Bool(true),
 		},
+		{
+			ParameterKey:   aws.String("CustomNodeSecurityGroups"),
+			ParameterValue: aws.String(strings.Join(input.SecurityGroups, ",")),
+		},
```

Comment:
**Required**: we **MUST** be able to handle user requests where the custom node security groups are meant to keep their values and not be updated.

For example when the user only wants to modify the image ID or the volume size they don't have to specify all the fields of `eks.NodePoolUpdate` in the update request, they can use a minimal update descriptor YAML like the following:
```yaml
image: "image-id-value"
```
or
```yaml
volumeSize: 21
```

AFAIK we would like to support the same optional behavior with `securityGroups` thus the stack update parameters should behave accordingly.
So the examples above should not modify the custom node security groups, those values should only be modified by explicit update requests containing the `securityGroups` field. 
You can use the available optional stack parameter constructor to achieve this.

(For the record: I would check for nil value as condition for using the existing stack parameter value and let the empty value unset the security groups while non-empty value sets them.)

```suggestion
		sdkCloudFormation.NewOptionalStackParameter(
			"CustomNodeSecurityGroups",
			input.SecurityGroups != nil,
			strings.Join(input.SecurityGroups, ","),
		),
```

Please also manually test this case thoroughly, especially the behaviors on unspecified `securityGroups`, no-item specified `securityGroups` and multiple item specified `securityGroups`.